### PR TITLE
[master] Porting #52502 to master

### PR DESCRIPTION
**Pull request by garethgreenaway**
Porting #52502 to master

### CODE REVIEW

**Comment by garethgreenaway** [comment]
re-run full all

**Review by dwoz — APPROVED** [review]

File: salt/states/archive.py [hunk 1]
```diff
@@ -1350,10 +1350,13 @@ def extracted(name,
                         )
                         return ret
 
-                    tar_opts = shlex.split(options)
+                    # Ignore verbose file list options as we are already using
+                    # "v" below in tar_shortopts
+                    tar_opts = [x for x in shlex.split(options)
+                                if x not in ('v', '-v', '--verbose')]
```
Comment (dwoz) [inline salt/states/archive.py]: What happens if someone passes `xv`? Should this check be something like:

```
if not x.startswith('-') and 'v' not in x
```

in addition to what it is checking now?

File: salt/states/archive.py [hunk 2]
```diff
@@ -1383,9 +1386,9 @@ def extracted(name,
                         ret['changes'] = results
                         return ret
                     if _is_bsdtar():
-                        files = results['stderr']
+                        files = results['stderr'].splitlines()
                     else:
-                        files = results['stdout']
+                        files = results['stdout'].splitlines()
```
Comment (dwoz) [inline salt/states/archive.py]: What happens if someone extracts an archive with 100K + files?

**Review by dwoz — CHANGES_REQUESTED** [review]

**Comment by dwoz** [comment]
Closing this due to inactivity.  Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

**Pull request closed without merging**